qcodes/instrument_drivers/Shamrock/SR750.py

```diff
@@ -0,0 +1,305 @@
+        c_grating = ctypes.c_int(grating)
+        code = self.dll.ShamrockSetGrating(c_device, c_grating)
+        self.error_check(code, 'ShamrockSetGrating')
+
+    def set_number_pixels(self, device, number_pixels):
+        c_device = ctypes.c_int(device)
+        c_number_pixels = ctypes.c_int(number_pixels)
+        code = self.dll.ShamrockSetNumberPixels(c_device, c_number_pixels)
+        self.error_check(code, 'ShamrockSetNumberPixels')
+
+    def set_pixel_width(self, device, width):
+        c_device = ctypes.c_int(device)
+        c_width = ctypes.c_float(width)
+        code = self.dll.ShamrockSetPixelWidth(c_device, c_width)
+        self.error_check(code, 'ShamrockSetPixelWidth')
+
+    def set_slit(self, device, width):
+        c_device = ctypes.c_int(device)
+        c_width = ctypes.c_float(width)
+        code = self.dll.ShamrockSetSlit(c_device, c_width)
+        self.error_check(code, 'ShamrockSetSlit')
+
+    def set_wavelength(self, device, wavelength):
+        c_device = ctypes.c_int(device)
+        c_wavelength = ctypes.c_float(wavelength)
+        code = self.dll.ShamrockSetWavelength(c_device, c_wavelength)
+        self.error_check(code, 'ShamrockSetWavelength')
+
+
+class Shamrock_SR750(Instrument):
+    """
+    Instrument driver for the Shamrock SR750 spectrometer.
+
+    Args:
+        name (str): Instrument name.
+        dll_path (str): Path to the ShamrockCIF.dll file. If not set, a default path is used.
+        device_id (int): ID for the desired spectrometer.
+        ccd_number_pixels (int): Number of pixels on the connected CCD.
+        ccd_pixel_width (int): Pixel width on the connected CCD.
+
+    Attributes:
+        ShamrockCIF (ShamrockCIF): DLL wrapper for ShamrockCIF.dll
+        device_id (int): Spectrometer device ID.
+        serial_number (int): Serial number of the spectrometer.
+        number_gratings (int): Number of gratings on the spectrometer.
+    """
+
+    def __init__(self, name, dll_path=None, device_id=0, ccd_number_pixels=1024, ccd_pixel_width=26, **kwargs):
+
+        super().__init__(name, **kwargs)
+
+        # link to dll
+        self.ShamrockCIF = ShamrockCIF(dll_path=dll_path)
+
+        # store device number
+        self.device_id = device_id
+
+        # initialize Shamrock
+        self.ShamrockCIF.initialize()
+
+        # read info from Shamrock
+        self.serial_number = self.ShamrockCIF.get_serial_number(self.device_id)
+        self.number_gratings = self.ShamrockCIF.get_number_gratings(self.device_id)
+
+        # send CCD info to Shamrock
+        self.ShamrockCIF.set_number_pixels(self.device_id, ccd_number_pixels)
+        self.ShamrockCIF.set_pixel_width(self.device_id, ccd_pixel_width)
+
+        # add the instrument parameters
+        self.add_parameter('blaze',
+                           get_cmd=self.get_blaze,
+                           get_parser=int,
+                           label='blaze')
+
+        self.add_parameter('calibration',
```

Comment:
this parameter is not settable, also, when gotten, it returns an array. In this situation, the question is - does "it" have to be a parameter? Why not having it just in the form of `get_calibration` method? Also, a related question, does the calibration array have to be stored in a snapshot (it's quite long, but does it provide value when included in the snapshot)?

Reply:
@SvenBo90 what is your take on this?